src/datatype/memory_bvecf32.rs

```diff
@@ -152,6 +153,38 @@ impl IntoDatum for BVecf32Output {
     }
 }
 
+impl FromDatum for BVecf32Output {
+    unsafe fn from_polymorphic_datum(datum: Datum, is_null: bool, _typoid: Oid) -> Option<Self> {
+        if is_null {
+            None
+        } else {
+            let ptr = unsafe {
+                NonNull::new(pgrx::pg_sys::pg_detoast_datum(
+                    datum.cast_mut_ptr::<BVecf32Header>().cast(),
+                ))
+            };
+            ptr.map(|ptr| BVecf32Output(ptr.cast()))
+        }
+    }
+}
+
+unsafe impl UnboxDatum for BVecf32Output {
+    type As<'src> = BVecf32Output;
+    #[inline]
+    unsafe fn unbox<'src>(d: pgrx::Datum<'src>) -> Self::As<'src>
+    where
+        Self: 'src,
+    {
+        let datum = d.sans_lifetime();
+        let ptr = unsafe {
```

Comment:
You must check whether the return value of `pg_detoast_datum` equals to the arguemnt to solve the ownership issues.